src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -146,6 +154,11 @@ protected internal override void SetPinMode(int pinNumber, PinMode mode)
             {
                 try
                 {
+                    if (File.ReadAllText(directionPath).Trim() == sysFsMode) // PIN direction already set
```

Comment:
I'm not sure this extra check is really saving us anything, since you are still creating the stream and checking it's contents, which probably takes equivalent time than re-writing the value. On the other hand, in the case this value is different we would be doubling the time for that, so I don't think that this extra check is really worth it.

Reply:
For my case, writing to the direction fill will throw an exception, because the direction cannot be changed. I just added this check because the exception message "requires root" was a bit confusing for me when running the app as root. So it should also be fine to not add this check since it is not necessary to use this method/constructor.

```
# echo "in" > /sys/class/gpio/gpio48/direction
-bash: /sys/class/gpio/gpio48/direction: Permission denied
```